kube/src/api/params.rs

```diff
@@ -0,0 +1,319 @@
+                "Force is applicable only for Apply strategy!".into(),
+            ))
+        } else {
+            Ok(())
+        }
+    }
+
+    pub(crate) fn populate_qp(&self, qp: &mut url::form_urlencoded::Serializer<String>) {
+        if self.dry_run {
+            qp.append_pair("dryRun", "true");
+        }
+        if self.force {
+            qp.append_pair("force", "true");
+        }
+        if let Some(ref field_manager) = self.field_manager {
+            qp.append_pair("fieldManager", &field_manager);
+        }
+    }
+
+    /// Construct PatchParams for server-side apply
+    pub fn apply(manager: &str) -> Self {
+        Self {
+            patch_strategy: PatchStrategy::Apply,
+            field_manager: Some(manager.into()),
+            ..Self::default()
+        }
+    }
+
+    /// Force the result through on conflicts
+    pub fn force(mut self) -> Self {
+        self.force = true;
+        self
+    }
+
+    /// Perform a dryRun only
+    pub fn dry_run(mut self) -> Self {
+        self.dry_run = true;
+        self
+    }
+}
+
+/// Four different patch types are supported.
+///
+/// Apply strategy is kinda special
+#[derive(Clone, PartialEq)]
+pub enum PatchStrategy {
+    /// [Server side apply](https://kubernetes.io/docs/reference/using-api/api-concepts/#server-side-apply)
+    ///
+    /// Requires kubernetes >=1.16
+    Apply,
+    /// A [JSON merge](https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-json-merge-patch-to-update-a-deployment)
+    JSON,
+    /// A regular merge
+    Merge,
+    /// A [stategic merge](https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-strategic-merge-patch-to-update-a-deployment)
+    Strategic,
+}
+
+impl std::fmt::Display for PatchStrategy {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let content_type = match &self {
+            Self::Apply => "application/apply-patch+yaml",
+            Self::JSON => "application/json-patch+json",
+            Self::Merge => "application/merge-patch+json",
+            Self::Strategic => "application/strategic-merge-patch+json",
+        };
+        f.write_str(content_type)
+    }
+}
+
+// Kubectl defaults to Strategic strategy, but doing so will break existing consumers
+// so, currently we still default to Merge it may change in future versions
+// Strategic merge doesn't work with CRD types https://github.com/kubernetes/kubernetes/issues/52772
+impl Default for PatchStrategy {
+    fn default() -> Self {
+        PatchStrategy::Merge
+    }
+}
+
+/// Common query parameters for delete calls
+#[derive(Default, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DeleteParams {
+    /// When present, indicates that modifications should not be persisted.
+    #[serde(
+        serialize_with = "dry_run_all_ser",
+        skip_serializing_if = "std::ops::Not::not"
+    )]
+    pub dry_run: bool,
```

Comment:
This is new, used to be Option<bool>, but there are different rules for body params.